haskell/experimental/private/module.bzl

```
@@ -236,8 +236,12 @@ def _build_haskell_module(
         moduleAttr.plugins,
         moduleAttr.tools,
     ]
+    if plugins and not enable_th:
+        # For #1681. These suppresses bogus warnings about missing libraries which
+        # aren't really needed.
```

```suggestion
        # For #1681. This suppresses bogus warnings about missing libraries which
        # aren't really needed.
```
Is there a GHC ticket about these? Bogus warnings seem like a compiler issue.

---

They are bogus in the context of `rules_haskell`. But if the user asks to have these missing libraries reported with `-Wmissing-extra-shared-lib`, I couldn't say the messages are bogus in general.

We could try to argue perhaps that the interpreter shouldn't try to load these libraries automatically. But it isn't clear to me either that a plugin can't request the interpreter to evaluate code that depends on these libraries. It would count pretty much as using TH, I guess.